pyccel/codegen/pipeline.py

```diff
@@ -325,41 +326,45 @@ def handle_error(stage):
                     # remove library folder to avoid missing files and copy
                     # new one from pyccel stdlib
                     lib_dest_path = os.path.join(pyccel_dirpath, lib_name)
-                    if os.path.exists(lib_dest_path):
-                        shutil.rmtree(lib_dest_path)
-                    shutil.copytree(lib_path, lib_dest_path)
+                    lock_path = lib_dest_path + '.lock'
+                    lock = FileLock(lock_path)
+                    with lock:
```

Comment:
If the variables `lock` and `lock_path` are not used anywhere else in this file, I would recommend that we condense these three lines into one, as you do on line 353:
```python
with FileLock(lib_dest_path + '.lock'):
```

Reply:
Beat you to it. See 96ae409 ;)

Reply:
:laughing: